lesson-3/js/main.js

```diff
@@ -1,65 +1,78 @@
 function palindrome(str) {
-    var strLen = str.length; //переменная, равная длине слова//
-    for (var i = 0; i < strLen; i++) {
-      if (str[i] === str[strLen - 1 - i]) {
-        return true; //сравниваем буквы с разных концов слова//
+  str = str.toLowerCase().replace(/[^а-яa-z]/);
+  //we should specify symbols which we check, in this case latin and cyrillic//
+  for (let i = 0; i < str.length; i++) {
+      if (str[i] !== str[str.length - 1 - i]) {
+          return false;
       }
-    }
-    return false;
   }
-  
-  test = palindrome("шалаш");
-  console.log(test);
-  
-  var a = 5; //задаем переменные//
-  var b = 8;
-  
-  function comparison(a, b) {
-    if (a < b) {
-      return a; //сравниваем их, выводится меньшая//
-    }
-    return b;
-  }
-  result = comparison(a, b);
-  console.log(result);
-  
-  function compare(a, b) { //то же, с тернарным оператором//
-    return a < b ? a : b
-  };
-  var a = 10;
-  var b = 4;
-  var f = compare(a, b);
-  console.log(f);
-  
-  
-  var a = 30;
-  var y = 38;
-  
-  function collation(a, y) { //поиск наибольшей из величин//
-    if (a > y) {
+  return true;
+}
+
+test = palindrome("wwerr");
+console.log(test);
+
+let a = 11;
+let b = 8;
+
+function comparison(a, b) {
+  if (a < b) {
       return a;
-    }
-    return y;
   }
-  result = collation(a, y);
-  console.log(result);
-  
-  
-  function collate(a, y) {
-    return a > y ? a : y
-  };
-  var a = 66;
-  var y = 48;
-  var f = collate(a, y);
-  console.log(f);
-  
-  
-  function replaceZero(number) { //решение из интернета, сама не сделала((//
-    const numStr = `${number}`;
-    if (numStr.includes('0')) {
+  return b;
+}
+result = comparison(a, b);
+console.log(result);
+
+function compare(p, t) {
+  return p < t ? p : t
+};
+const p = 10;
+const t = 15;
+const f = compare(p, t);
+console.log(f);
+
+
+const z = 30;
+const y = 38;
+
+function collation(z, y) {
+  if (z > y) {
+      return z;
+  }
+  return y;
+}
+result = collation(z, y);
+console.log(result);
+
+
+function collate(c, d) {
+  return c > d ? c : d
+};
+const c = 66;
+const d = 78;
+const n = collate(c, d);
+console.log(n);
+
+
+function replaceZero(number) {
+  const numStr = `${number}`;
+  if (numStr.includes('0')) {
       return `${number}`.replaceAll('0', 'zero');
-    }
-    return number;
   }
-  
-  console.log([12, 53, 20, 18, 22, 100, 43, 53, 50].map(replaceZero).join(', '));
\ No newline at end of file
+  return number;
+}
+
+console.log([12, 53, 20, 18, 22, 100, 43, 53, 50].map(replaceZero).join(', '));
+
+function replaceZero(number) {
+  /*if we use numStr instead of `${number}`, how should we define it?
+  I don't understand*/
+  const numStr = " ";
+  if (numStr.includes('0')) {
+      return numStr.replaceAll('0', 'zero');
+  }
+  return number;
+}
```

Comment:
I meant to replace it with a variable, since we have it
![image](https://user-images.githubusercontent.com/20189313/132668933-8e828dab-de18-49be-8b1a-a10a0bcea61a.png)